libs/partners/openai/langchain_openai/embeddings/base.py

```diff
@@ -307,106 +285,106 @@ def _invocation_params(self) -> Dict[str, Any]:
-                    indices.append(i)
+            return self._transformers_tokenize_and_split(texts)
         else:
-            try:
-                encoding = tiktoken.encoding_for_model(model_name)
-            except KeyError:
-                encoding = tiktoken.get_encoding("cl100k_base")
-            encoder_kwargs: Dict[str, Any] = {
-                k: v
-                for k, v in {
-                    "allowed_special": self.allowed_special,
-                    "disallowed_special": self.disallowed_special,
-                }.items()
-                if v is not None
-            }
-            for i, text in enumerate(texts):
-                if self.model.endswith("001"):
-                    # See: https://github.com/openai/openai-python/
-                    #      issues/418#issuecomment-1525939500
-                    # replace newlines, which can negatively affect performance.
-                    text = text.replace("\n", " ")
-
-                if encoder_kwargs:
-                    token = encoding.encode(text, **encoder_kwargs)
-                else:
-                    token = encoding.encode_ordinary(text)
+            return self._tiktoken_tokenize_and_split(texts)
 
-                # Split tokens into chunks respecting the embedding_ctx_length
-                for j in range(0, len(token), self.embedding_ctx_length):
-                    tokens.append(token[j : j + self.embedding_ctx_length])
-                    indices.append(i)
-
-        if self.show_progress_bar:
-            try:
-                from tqdm.auto import tqdm
+    def _transformers_tokenize_and_split(
+        self, texts: List[str]
+    ) -> Tuple[List[Union[List[int], str]], List[int]]:
+        tokens: List[Union[List[int], str]] = []
+        indices: List[int] = []
+        model_name = self.tiktoken_model_name or self.model
 
-                _iter: Iterable = tqdm(range(0, len(tokens), chunk_size))
-            except ImportError:
-                _iter = range(0, len(tokens), chunk_size)
-        else:
-            _iter = range(0, len(tokens), chunk_size)
-        return _iter, tokens, indices
+        try:
+            from transformers import AutoTokenizer
+        except ImportError:
+            raise ValueError(
+                "Could not import transformers python package. "
+                "This is needed in order to for OpenAIEmbeddings without "
+                "`tiktoken`. Please install it with `pip install transformers`. "
+            )
 
-    # please refer to
+        tokenizer = AutoTokenizer.from_pretrained(
+            pretrained_model_name_or_path=model_name
+        )
+        for i, text in enumerate(texts):
+            # Tokenize the text using HuggingFace transformers
+            tokenized: List[int] = tokenizer.encode(text, add_special_tokens=False)
+
+            # Split tokens into chunks respecting the embedding_ctx_length
+            for j in range(0, len(tokenized), self.embedding_ctx_length):
```

Comment:
I _think_ if you pass in an empty text, this will do weird things with the number of entries. Not 100% confident though